src/shell/shell.cpp

```diff
@@ -691,10 +710,12 @@ void SHELL_Init() {
 	MSG_Add("SHELL_CMD_SUBST_HELP","Assign an internal directory to a drive.\n");
 	MSG_Add("SHELL_CMD_LOADHIGH_HELP","Loads a program into upper memory (requires xms=true,umb=true).\n");
 
-	MSG_Add("SHELL_CMD_LS_HELP", "List directory contents.\n");
-	MSG_Add("SHELL_CMD_LS_HELP_LONG", "ls [/?] [PATTERN]\n");
-	MSG_Add("SHELL_CMD_LS_PATH_ERR",
-	        "ls: cannot access '%s': No such file or directory\n");
+	MSG_Add("SHELL_CMD_LS_HELP", "Lists directory contents.\n");
+	MSG_Add("SHELL_CMD_LS_HELP_LONG", "LS [drive:][path][filename] [/A] [/L] [/P] [/Z]\n\n"
+	        "  /A\tLists hidden and system files also.\n"
```

Comment:
Can you expand on how `/A` is supposed to be implemented? I think it will be extremely hard to correctly implement this in a cross-platform manner.

Reply:
There are different drive types in DOSBox-staging/-X such as local drives and FAT drives. For FAT drives (i.e. mounted disk images), these attributes can be fully implemented for all platforms. For local drives, the implementation as in DOSBox-X is this: for Windows platforms fully attributes are implemented, and for other platforms only the archive and read-only attributes are implemented. As for other drives (virtual and ISO drives), they are read-only drives so the file attributes are read as is from them.